client/src/components/NewReviewForm.js

```diff
@@ -0,0 +1,116 @@
+import React, {useState } from "react"
+import ErrorList from "./layout/ErrorList"
+
+const NewReviewForm = ({ pizzaId, pizzaPlace, setPizzaPlace, currentUser }) => {
+  const [newReview, setNewReview] = useState({
+    userId: currentUser.id,
+    rating: 3,
+    title: "",
+    text: ""
+  })
+
+  const [errors, setErrors] = useState({})
+
+  const postNewReview = async(newReviewData) => {
+    try {
+      const response = await fetch(`/api/v1/pizza-places/${pizzaId}/reviews/new`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(newReviewData)
+      })
+      if (!response.ok) { 
+        if (response.status === 422){
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        } else {
+          throw new Error(`${response.status} (${response.statusText})`)
+        }
+      } else {
+        const body = await response.json()
+        return body.newPizzaReview
+      }
+    } catch(error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  const handleInputChange = event => {
+    setNewReview({
+      ...newReview,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    const reviewData = await postNewReview(newReview)
+    setPizzaPlace({
+      ...pizzaPlace,
+      reviews: [...pizzaPlace.reviews, reviewData]
+    })
+    clearForm()
+  }
+
+  const clearForm = () => {
+    setNewReview({
+      userId: currentUser.id,
```

Comment:
should not be keeping track of the user in state - the application already knows who the user is through the passport session in the back-end